server/handlers/find.js

```diff
@@ -25,188 +28,220 @@ export const findByCredential = async user => {
     error.password = 'Invalid password';
     return Promise.reject(error);
   }
-  return foundUser;
+  return Promise.resolve(foundUser);
 };
 
 /** Returns a user that matches _id
  * @param {string} _id
  * @returns {Promise<User|UserNotFound>}
  */
 export const findById = async _id => {
-  const error = {};
-  const foundUser = await User.findOne({
+  return await User.findOne({
     _id,
+  }).catch(err => {
+    console.error(err);
+    return Promise.reject(err);
   });
-  if (!foundUser) {
-    error.user = 'User not found';
-    return Promise.reject(error);
-  }
-  return foundUser;
 };
 
 /** Returns a user that matches handle
  * @param {string} userHandle
  * @returns {Promise<User|UserNotFound>}
  */
 export const findByHandle = async userHandle => {
-  const error = {};
   const foundUser = await User.findOne({
     userHandle,
+  }).catch(err => {
+    console.error(err);
+    return Promise.reject(err);
   });
   if (!foundUser) {
-    error.user = 'User not found';
-    return error;
+    return Promise.reject({ user: 'User not found' });
   }
-  return foundUser;
+  return Promise.resolve(foundUser);
```

Comment:
in an async function, you don't need Promise.resolve, you can just return directly and it will work.

Reply:
Fixed